src/libs/Navigation/AppNavigator/AuthScreens.js

```diff
@@ -210,139 +216,112 @@ class AuthScreens extends React.Component {
             chatShortcutConfig.modifiers,
             true,
         );
-    }
 
-    shouldComponentUpdate(nextProps) {
-        return nextProps.windowHeight !== this.props.windowHeight || nextProps.isSmallScreenWidth !== this.props.isSmallScreenWidth;
```

Comment:
This is root component and it's important to prevent unnecessary re-rendering.
So let's use `memo` which is equivalent to `shouldComponentUpdate` in function component

Reply:
In function components we should use hooks instead of HOCs, so there is no information about windowHeight & isSmallScreenWidth in props. I've checked re-rendering and we don't have regression in this field. Please notice, that the only one dependency we have is `isSmallScreenWidth` and the component will re-render only if that variable has been changed.
This optimization is no longer needed :)

Reply:
> so there is no information about windowHeight & isSmallScreenWidth in props

We can access in `memo`
i.e.
https://github.com/Expensify/App/blob/d4a33cd5a2f21d2ee853682f8938a965b43c52b9/src/components/PopoverWithMeasuredContent.js#L164-L169

Reply:
@barttom bump ^

We don't need to re-render this component when seesion, lastOpenedPublicRoomID, isUsingMemoryOnlyKeys, lastUpdateIDAppliedToClient Onyx prop values are changed as they're used only on component mount.

Reply:
@situchan sorry for not responding. I’m OOO until end of this week too.

I used hook instead for f HOC so I don’t have access to the dimensions props.
Using hooks is more optimal and previous shouldComponentUpdate is not needed anymore

Reply:
I am not suggesting HOC. I am not sure what you're talking about.

Reply:
On first comment You suggest using memo and rewrite previous implementation of componentShouldUpdate. Without HOC we don’t have dimensions props anymore.

Reply:
Of course, if using `useDimensions`, component is always re-rendered. So we can simply return false in memo function so that other props changes don't cause component re-render.

Reply:
@situchan memoization added